src/lib/battery/battery.cpp

```diff
@@ -239,10 +239,10 @@ float Battery::calculateStateOfChargeVoltageBased(const float voltage_v, const f
 void Battery::estimateStateOfCharge()
 {
 	// choose which quantity we're using for final reporting
-	if (_params.capacity > 0.f && _battery_initialized) {
+	if ((_params.capacity > 0.f) && _battery_initialized && (_state_of_charge_volt_based > _params.emergen_thr)) {
```

Comment:
This jump could be a bit weird if you're on the line? Maybe it needs to latch or have a big hysteresis?

Reply:
Agree. Maybe it's enough to go with the filter adjustments below?

Reply:
> This jump could be a bit weird if you're on the line?

I agree but it's only there to save the user from crashing the vehicle after he already has a seriously misconfigured battery estimation.
We can try to be super smart and make it unnoticeable that the capacity was set wrong but what was the point of having the parameter in the first place? I expected we can rely on it being correct if it's set.

> Maybe it's enough to go with the filter adjustments below?

If the capacity is set completely wrong like in some logs I saw it will still react too slow and the vehicle might crash even though "it can be seen from the voltage that it was too low".